dissect/target/tools/info.py

```diff
@@ -0,0 +1,140 @@
+from dissect.target.tools.query import record_output
+from dissect.target.tools.utils import (
+    configure_generic_arguments,
+    process_generic_arguments,
+)
+
+InfoRecord = TargetRecordDescriptor(
+    "target/info",
+    [
+        ("datetime", "last_activity"),
+        ("datetime", "install_date"),
+        ("net.ipaddress[]", "ips"),
+        ("string", "os_family"),
+        ("string", "os_version"),
+        ("string", "architecture"),
+        ("string", "language"),
+        ("string", "timezone"),
+    ],
+)
+
+
+log = logging.getLogger(__name__)
+logging.lastResort = None
+logging.raiseExceptions = False
+
+
+def main():
+    help_formatter = argparse.ArgumentDefaultsHelpFormatter
+    parser = argparse.ArgumentParser(
+        description="target-info",
+        fromfile_prefix_chars="@",
+        formatter_class=help_formatter,
+        add_help=False,
+    )
+    parser.add_argument("targets", metavar="TARGETS", nargs="*", help="Targets to display info from")
+    parser.add_argument("--from-file", nargs="?", type=Path, help="file containing targets to load")
+    parser.add_argument("-d", "--delimiter", default=" ", action="store", metavar="','")
+    parser.add_argument("-s", "--strings", action="store_true", help="print output as string")
+    parser.add_argument("-r", "--record", action="store_true", help="print output as record")
+    parser.add_argument("-j", "--json-pretty", action="store_true", help="output records as pretty json")
+    parser.add_argument("-J", "--json", action="store_true", help="output records as one-line json")
+    configure_generic_arguments(parser)
+
+    args = parser.parse_args()
+
+    process_generic_arguments(args)
+
+    if not args.targets and not args.from_file:
+        parser.error("too few arguments")
+
+    if args.from_file:
+        if not args.from_file.exists():
+            parser.error(f"--from-file {args.from_file} does not exist")
+
+        targets = args.from_file.read_text().splitlines()
+        while targets[-1] == "":
+            targets = targets[:-1]
+        args.targets = targets
+
+    for target in Target.open_all(args.targets):
+        if args.json:
+            print(json.dumps(obj_target_info(target)))
+        elif args.json_pretty:
+            print(json.dumps(obj_target_info(target), indent=4))
+        elif args.record:
+            rs = record_output(args.strings, args.json)
+            rs.write(
+                InfoRecord(
+                    last_activity=target.activity,
+                    install_date=target.install_date,
+                    ips=target.ips,
+                    os_family=target.os,
+                    os_version=target.version,
+                    architecture=target.architecture,
+                    language=target.language,
+                    timezone=target.timezone,
+                    _target=target,
+                )
+            )
+        else:
+            print_target_info(target)
+
```

Comment:
Maybe put this code into a try/except with an error, such as:
```python
target.log.error("Exception in retrieving information for target: `%s`", target, exc_info=True)
```

Reply:
Added in [94af2a6](https://github.com/fox-it/dissect.target/pull/136/commits/94af2a62f1b38dd7c6046bdb27d7a94ccc82aa00).